package.json

```diff
@@ -53,8 +54,10 @@
     "build-css": "node-sass src/ -o src/",
     "watch-css": "npm run build-css && node-sass src/ -o src/ --watch --recursive",
     "start-js": "react-scripts start",
-    "start": "npm-run-all -p build watch-css start-js",
-    "build": "npm run build-css && npm run copy-monaco && react-scripts build",
+    "start": "env-cmd --fallback ./.env.development npm-run-all -p build watch-css start-js",
```

Comment:
@storrisi  I see what you have done here. The idea of using --fallback so that when running `npm start`  the configuration will be read from the **.env** default file if no  specific **.env.<environment>** file is present. Good thinking, but I think that if the user needs to run locally with different configurations other than "development" or "default" we might end up with a few "complications".

@jeffmcaffer watch-css is used when starting a local env, but when deploying on environments I do expect that just the **build** command will be executed or an env specific instruction, if we follow the **, build:env** approach suggested by @storrisi .


Anyways, I would rather think about this implementation from a different angle, and provide a way for both a user and a CI/CD process to provide any .env configuration without having to launch a different **npm script** at each time.

I think that ideally, we could proceed with one of the following 2 approaches:
###  Use only a .env file when running builds 
we could have only one build/start script reading configurations only and exclusively from the default **.env** file. A developer can then run a specific build as follows:

- Copy the **sample.env** configuration into a local **.env** file
- Change the configuration according to the desired one (Production APIs, Development etc)
- Run **npm start** which will build and serve the app reading from the **.env** file
Similarly, our **CI/CD** process can generate the **.env** file, by reading it from an **environment specific** file or directly from the CI SECRETS configuration, Copy over the given configuration, run build + deployment on any given envinroment.  Long story short, the CI/CD always injects the **development** or **production** config into a **.env** file, hence triggers the build

### Modify the NPM build & start scripts  so that external arguments are provided
We could alternatively change the npm scripts to pass a specific ".env.environment"  file externally
The difference in the previous approach would consist in the developer and the CI  launching**npm start** and **build** scripts providing the name of the specific **.env.development ** or **.env.anything** files to be used.

- ### use 1 .rc file
see https://www.npmjs.com/package/env-cmd#rc-file-usage

Given the --fallback instruction, the result would be a merge  
 

 


**NOTE:** While we can commit samples/ public configurations on the repository, so developers know how to setup their **.env** file (they can just copy-paste and fill in the specific conf), we should still keep the **.env** or **.env.environment** files outside the repo. This would give us the flexibility to keep the same approach when introducing new configurations like for example API_ACCESS_KEYS, and similar.


 @storrisi @jeffmcaffer thoughts?